app/controllers/api/base_controller.rb

```diff
@@ -1,26 +1,78 @@
 class Api::BaseController < ApplicationController
   class SchemaValidationError < StandardError ; end
 
-  def render_schema_json(data, schema: nil, **kwargs)
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def render_schema_json(data, **kwargs)
+    data = resource_for_json_rendering(data)
+
     if Rails.env.local? && !(defined?(Runger.config) && Runger.config.skip_schema_validation?)
-      # TIP: Generate the schema from sample response data using https://app.quicktype.io/ .
+      # TIP: Generate the schema from sample response data using
+      # https://jsonformatter.org/json-to-jsonschema .
       partial_schema_path =
         schema ||
         "#{params[:controller]}/#{params[:action]}".delete_prefix('api/')
-      full_schema_path = "spec/support/schemas/#{partial_schema_path}.json"
+
+      full_schema_path =
+        Rails.root.join(
+          "spec/support/schemas/#{partial_schema_path}.json",
+        ).to_s
+
       schema_validation_errors =
-        JSON::Validator.fully_validate(
-          full_schema_path,
-          data.to_json,
-          strict: true,
-          validate_schema: true,
-        )
+        begin
+          JSON::Validator.fully_validate(
+            full_schema_path,
+            data.is_a?(String) ? data : data.to_json,
+            strict: true,
+            validate_schema: true,
+            clear_cache: true,
```

Comment:
`clear_cache: true` is necessary to see changes in development without restarting the server.